guis/panel/leak/PlotLeakRate.py

```diff
@@ -231,9 +232,15 @@ def FitAndPlot_2(df, axDiffP):
 ################################################################################
 # main
 ################################################################################
-def main(options):
+def main(options, panel=None, tag=None):
     # read raw data files into a dataframe
-    df = ReadLeakRateFile(options.infile, options.is_new_format)
+    if isinstance(options, pd.DataFrame):
```

Comment:
This overloading of `options` and comparison with `isinstance` is not good practice. It should be clear from a function's interface what types of objects you could pass. This would absolutely not fly in just about any other language.

See [here](https://stackoverflow.com/questions/3111611/is-this-use-of-isinstance-pythonic-good) for discussion.

It's coming from a good place, though. Indeed this function feels wrong, doesn't it? Originally, it was two functions, but that was bad because there was a ton of code duplication. Now that it's one function, we have too many if's and checks.

There's a way to solve this. Gotta think about it. It's in [here](https://en.wikipedia.org/wiki/Design_Patterns) somewhere...

Reply:
Without getting fancy with design patterns, can we just shove everything into the options?

Reply:
https://en.wikipedia.org/wiki/Strategy_pattern